WcaOnRails/app/models/competition.rb

```diff
@@ -946,15 +935,18 @@ def uses_cumulative_across_rounds?
   end
 
   # The name `is_probably_over` is meant to be surprising.
-  # We don't actually know when competitions are over, because we don't know their schedules, nor
-  # do we know their timezones.
-  # See discussion here: https://github.com/thewca/worldcubeassociation.org/pull/1206/files#r98485399.
+  # While we now know competition schedules, not all competitions might have a schedule
   def is_probably_over?
-    !end_date.nil? && end_date < Date.today
+    last_event_end = top_level_activities.map { |a| a.end_time }.max
```

Comment:
Do you know what sort of an effect this has on the performance of the `/competitions` page? Is this going to be an `n+1` query?